variables.tf

```
@@ -104,8 +116,33 @@ variable "override_policy_documents" {
 
 variable "policy_statements" {
   description = "A list of IAM policy [statements](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document#statement) for custom permission usage"
-  type        = any
-  default     = []
+  type = list(object({
```

I messed up and didn't keep it consistent in some other modules, but we want this as a `map(object{...})`

```suggestion
  type = map(object({
```